src/python/zquantum/core/wip/estimators/estimation.py

```diff
@@ -141,39 +141,54 @@ def proportional_shot_allocation(
     if total_n_shots <= 0:
         raise ValueError("total_n_shots must be positive.")
 
-    def _allocate_shots(
-        estimation_tasks: List[EstimationTask],
-    ) -> List[EstimationTask]:
-        frame_operators = [
-            estimation_task.operator for estimation_task in estimation_tasks
-        ]
+    frame_operators = [estimation_task.operator for estimation_task in estimation_tasks]
 
-        _, _, measurements_per_frame = estimate_nmeas_for_frames(
-            frame_operators, prior_expectation_values
-        )
+    _, _, measurements_per_frame = estimate_nmeas_for_frames(
+        frame_operators, prior_expectation_values
+    )
 
-        measurements_per_frame = scale_and_discretize(
-            measurements_per_frame, total_n_shots
+    measurements_per_frame = scale_and_discretize(measurements_per_frame, total_n_shots)
+
+    return [
+        EstimationTask(
+            operator=estimation_task.operator,
+            circuit=estimation_task.circuit,
+            number_of_shots=number_of_shots,
+        )
+        for estimation_task, number_of_shots in zip(
+            estimation_tasks, measurements_per_frame
         )
+    ]
 
-        return [
-            EstimationTask(
-                operator=estimation_task.operator,
-                circuit=estimation_task.circuit,
-                number_of_shots=number_of_shots,
-            )
-            for estimation_task, number_of_shots in zip(
-                estimation_tasks, measurements_per_frame
-            )
-        ]
 
-    return _allocate_shots
+def evaluate_estimation_circuits(
+    estimation_tasks: List[EstimationTask],
+    symbols_maps: List[List[Tuple[sympy.Basic, float]]],
+) -> List[EstimationTask]:
+    """
+    Evaluates circuits given in all estimation tasks using the given symbols_maps. If one symbols map is given,
+    it is used to evaluate all circuits. Otherwise, the symbols map at index i will be used for the estimation
+    task at index i.
+
+    Args:
+        estimation_tasks: the estimation tasks which contain the circuits to be evaluated
+        symbols_maps: a list of dictionaries (or singular dictionary) that map the symbolic symbols used in the
+            parametrized circuits to the associated values
+    """
+    return [
+        EstimationTask(
+            operator=estimation_task.operator,
+            circuit=estimation_task.circuit.evaluate(symbols_map),
+            number_of_shots=estimation_task.number_of_shots,
+        )
+        for estimation_task, symbols_map in zip(estimation_tasks, symbols_maps)
+    ]
 
 
-def naively_estimate_expectation_values(
+def estimate_expectation_values_by_averaging(
```

Comment:
I don't like `by_averaging` name.
I think the `BasicEstimator` was a good name, so I tried to emulate that with `naively`. 
What about `basic_estimate_expectation_values`, `standard_estimate_expectation_values`or sth like that?

Reply:
I respectfully disagree. 

`BasicEstimator` (and similarly `basic_estimate...` and `standard_estimate...`) doesn't really give any indication as to what is being done. It's only "basic" since averaging is what is considered commonplace in the industry. `by_averaging` on the other hand, explicitly communicates what is being done.